### PR TITLE
asXml adds a version declaration, remove it when extracting content

### LINKED ISSUE #6786: Import XLIFF in documents working add XML VERSION tag to some strings
Importing sometimes results in adding an XML version tag to the content:
version_tag

Resulting in:
broken_import_result

I don't know why exactly this happens, I run low on time and can't investigate it right now.

### PR DESCRIPTION
**Pull request by msphn**
This depending on the read string, it's not always the case.

Resolves #6786

@weisswurstkanone would you mind to share your opinion?

### CODE REVIEW

**Comment by weisswurstkanone** [comment]
Could you please describe a case + file which can be reproduced on the demo instance? Thx!

**Comment by msphn** [comment]
@weisswurstkanone actually no because the import fails. That's the bug I wrote the other patch for.

Screenshot 2020-07-16 at 12 15 14

I tried that file:

```
<?xml version="1.0" encoding="UTF-8"?>
<xliff version="1.2"><file original="document-136" source-language="en" target-language="de" datatype="html" tool="pimcore" category="document"><header/><body><trans-unit id="property~-~navigation_title"><source xml:lang="en"><![CDATA[Oldimer Cars]]></source></trans-unit><trans-unit id="property~-~navigation_name"><source xml:lang="en"><![CDATA[Home]]></source><target state="translated">DE - Home</target></trans-unit><trans-unit id="tag~-~cHeadline_0"><source xml:lang="en"><![CDATA[Oldtimer zum Verkauf!]]></source><target xml:lang="de" state="translated"><![CDATA[Neue Übersetzung]]></target></trans-unit><trans-unit id="tag~-~cText_0"><source xml:lang="en"><![CDATA[Mehr als 61.300 Oldtimer Autos und Trucks stehen zum Verkauf! ]]></source><target xml:lang="de" state="translated">Test test</target></trans-unit><trans-unit id="tag~-~cLink_0"><source xml:lang="en"><![CDATA[Finde deinen Oldtimer]]></source><target xml:lang="de" state="translated"><![CDATA[Find einen ganz tollen oldtimer]]></target></trans-unit><trans-unit id="tag~-~cHeadline_1"><source xml:lang="en"><![CDATA[Deine tägliche Dosis]]></source><target xml:lang="de" state="translated"><![CDATA[Deine tägliche Dosis Auto]]></target></trans-unit><trans-unit id="tag~-~cText_1"><source xml:lang="en"><![CDATA[Unser preisgekröntes Magazin ist der richtige Ort für die letzten News und Informationen]]></source><target xml:lang="de"><![CDATA[Unser preisgekröntes Magazin ist der richtige Ort für die letzten News und Informationen]]></target></trans-unit><trans-unit id="tag~-~cLink_1"><source xml:lang="en"><![CDATA[Mehr dazu]]></source><target xml:lang="de"><![CDATA[Mehr dazu]]></target></trans-unit><trans-unit id="tag~-~cHeadline_2"><source xml:lang="en"><![CDATA[Du willst mehr?]]></source><target xml:lang="de"><![CDATA[Du willst mehr?]]></target></trans-unit><trans-unit id="tag~-~cText_2"><source xml:lang="en"><![CDATA[Falls du nicht genug kriegen kannst, haben wir noch sehr viel mehr in der Pimcore Demo zu bieten]]></source><target xml:lang="de"><![CDATA[Falls du nicht genug kriegen kannst, haben wir noch sehr viel mehr in der Pimcore Demo zu bieten]]></target></trans-unit><trans-unit id="tag~-~cLink_2"><source xml:lang="en"><![CDATA[Mehr dazu]]></source><target xml:lang="de"><![CDATA[Mehr dazu]]></target></trans-unit><trans-unit id="tag~-~content:5.infoHeading"><source xml:lang="en"><![CDATA[61.569 Oldtimer stehen zum Verkauf!]]></source><target xml:lang="de"><![CDATA[61.569 Oldtimer stehen zum Verkauf!]]></target></trans-unit><trans-unit id="tag~-~content:5.infoSummary"><source xml:lang="en"><![CDATA[Unsere Mission ist den An- und Verkauf von Oldtimern für private Käufer und Verkäufer als auch Händler durch den Einsatz von exzellenten Tools zum Erlebnis zu machen. Wir sind ein lebendiger Demo-Marktplatz, der auf innovativer Spitzentechnologie aufbaut und sich ständig weiterentwickelt, um nicht nur den Bedürfnissen von Oldtimer-Enthusiasten gerecht zu werden. Zusätzlich bieten wir eine große Auswahl an Ersatzteilen für viele Marken und Modelle an.]]></source><target xml:lang="de"><![CDATA[Unsere Mission ist den An- und Verkauf von Oldtimern für private Käufer und Verkäufer als auch Händler durch den Einsatz von exzellenten Tools zum Erlebnis zu machen. Wir sind ein lebendiger Demo-Marktplatz, der auf innovativer Spitzentechnologie aufbaut und sich ständig weiterentwickelt, um nicht nur den Bedürfnissen von Oldtimer-Enthusiasten gerecht zu werden. Zusätzlich bieten wir eine große Auswahl an Ersatzteilen für viele Marken und Modelle an.]]></target></trans-unit><trans-unit id="tag~-~content:5.infoLinkPrimary"><source xml:lang="en"><![CDATA[Zur Suche]]></source><target xml:lang="de"><![CDATA[Zur Suche]]></target></trans-unit><trans-unit id="tag~-~content:5.infoLinkSecondary"><source xml:lang="en"><![CDATA[Zu den News]]></source><target xml:lang="de"><![CDATA[Zu den News]]></target></trans-unit><trans-unit id="tag~-~content:2.standardTeaserHeading"><source xml:lang="en"><![CDATA[Die zentrale Anlaufstelle für Oldtimer]]></source><target xml:lang="de"><![CDATA[Die zentrale Anlaufstelle für Oldtimer]]></target></trans-unit><trans-unit id="tag~-~content:2.title2"><source xml:lang="en"><![CDATA[Aktuelle Nachrichten]]></source><target xml:lang="de"><![CDATA[Aktuelle Nachrichten]]></target></trans-unit><trans-unit id="tag~-~content:2.text2"><source xml:lang="en"><![CDATA[Hole dir die aktuellsten Nachrichten, aufbereitet von unseren erstklassigen internationalen Journalisten.]]></source><target xml:lang="de"><![CDATA[Hole dir die aktuellsten Nachrichten, aufbereitet von unseren erstklassigen internationalen Journalisten.]]></target></trans-unit><trans-unit id="tag~-~content:2.link2"><source xml:lang="en"><![CDATA[Mehr dazu]]></source><target xml:lang="de"><![CDATA[Mehr dazu]]></target></trans-unit><trans-unit id="tag~-~content:4.heroGridBlock:1.heroGridHeadline1"><source xml:lang="en"><![CDATA[Chevrolet]]></source><target xml:lang="de"><![CDATA[Chevrolet]]></target></trans-unit><trans-unit id="tag~-~content:4.heroGridBlock:1.heroGridSubheadline1"><source xml:lang="en"><![CDATA[1970's Collection]]></source><target xml:lang="de"><![CDATA[1970's Collection]]></target></trans-unit><trans-unit id="tag~-~content:4.heroGridBlock:2.heroGridHeadline2"><source xml:lang="en"><![CDATA[Ford]]></source><target xml:lang="de"><![CDATA[Ford]]></target></trans-unit><trans-unit id="tag~-~content:4.heroGridBlock:2.heroGridSubheadline2"><source xml:lang="en"><![CDATA[Muscle Collection]]></source><target xml:lang="de"><![CDATA[Muscle Collection]]></target></trans-unit><trans-unit id="tag~-~content:4.heroGridBlock:3.heroGridHeadline3"><source xml:lang="en"><![CDATA[volkswagen]]></source><target xml:lang="de"><![CDATA[volkswagen]]></target></trans-unit><trans-unit id="tag~-~content:4.heroGridBlock:3.heroGridSubheadline3"><source xml:lang="en"><![CDATA[Classics for sale]]></source><target xml:lang="de"><![CDATA[Classics for sale]]></target></trans-unit><trans-unit id="tag~-~content:4.heroGridBlock:4.heroGridHeadline4"><source xml:lang="en"><![CDATA[vw VANS]]></source><target xml:lang="de"><![CDATA[vw VANS]]></target></trans-unit><trans-unit id="tag~-~content:4.heroGridBlock:4.heroGridSubheadline4"><source xml:lang="en"><![CDATA[Iconic Vintage Minibuses]]></source><target xml:lang="de"><![CDATA[Iconic Vintage Minibuses]]></target></trans-unit><trans-unit id="tag~-~content:14.productTeaserTitle"><source xml:lang="en"><![CDATA[Deine persönlichen Empfehlungen]]></source><target xml:lang="de"><![CDATA[Deine persönlichen Empfehlungen]]></target></trans-unit><trans-unit id="tag~-~content:11.infoHeading"><source xml:lang="en"><![CDATA[100% garantierte Kundenzufriedenheit]]></source><target xml:lang="de"><![CDATA[100% garantierte Kundenzufriedenheit]]></target></trans-unit><trans-unit id="tag~-~content:11.infoSummary"><source xml:lang="en"><![CDATA[Wir sind fest entschlossen, all unseren Kunden und Partnern den bestmöglichen Service zu bieten. Lese was unsere Händler, Privatverkäufer und Käufer über unseren Demo-Oldtimermarkt zu sagen haben.]]></source><target xml:lang="de"><![CDATA[Wir sind fest entschlossen, all unseren Kunden und Partnern den bestmöglichen Service zu bieten. Lese was unsere Händler, Privatverkäufer und Käufer über unseren Demo-Oldtimermarkt zu sagen haben.]]></target></trans-unit></body></file></xliff>

```

**Comment by weisswurstkanone** [comment]
IMHO the sample is wrong. The target is missing.

image

**Comment by msphn** [comment]
Not sure, I didn't read the spec to be honest. But it's what I patched in pimcore to cope with. I'm in meetings right now, I can try again tomorrow.

**Comment by msphn** [comment]
@weisswurstkanone 
I added a target tag and run it on demo.

Screenshot 2020-07-17 at 12 24 36

**Comment by msphn** [comment]
If you don't like my approach I could probably replace the current implementation of the value extraction with xpath. This would just do the job for now.

**Comment by msphn** [comment]
Just tell me what I should do and any chance we could get this done with 6.7?

I promised someone that pimcore offers great XLIFF support 😅

File: lib/Translation/Escaper/Xliff12Escaper.php [hunk 1]
```diff
@@ -116,6 +116,9 @@ private function parseInnerXml(string $content)
 
         $content = $node->asXml();
```
Comment (weisswurstkanone) [inline lib/Translation/Escaper/Xliff12Escaper.php:117]: please try to replace asXML with

$content = (string) $node

and remove your preg_replace again.

Let me know if this works.

**Comment by msphn** [comment]
@weisswurstkanone that works as well! I adjusted the PR

**Comment by weisswurstkanone** [comment]
Shouldn't this go in the 6.6 branch?

**Comment by msphn** [comment]
My other fixes are in 6.7 - is out soon anyway :)

**Comment by weisswurstkanone** [comment]
thanks a lot!

**Pull request merged**